src/utils/cardUtilityFunctions.js

```diff
@@ -122,3 +124,403 @@ export const getUpdatedCardSize = oldSize => {
   }
   return newSize;
 };
+
+// re-usable variable fetcher that returns an array of variables
+// variables are identified by surrounding curly braces i.e. {deviceid}
+export const getVariables = value => {
+  let variables = value && typeof value === 'string' ? value.match(/{[a-zA-Z0-9_-]+}/g) : null;
+  if (variables) {
+    variables = variables.map(variable => variable.replace(/[{}]/g, ''));
+  }
+  return variables;
+};
+
+// replace variables from the list of variables that are found on the target with their corresponding value
+export const replaceVariables = (variables, values, target) => {
+  let updatedTarget = target;
+  variables.forEach(variable => {
+    const variableRegex = new RegExp(`{${variable}}`, 'g');
+    updatedTarget = values[variable]
+      ? updatedTarget.replace(variableRegex, values[variable])
+      : target;
+  });
+  return updatedTarget;
+};
+
+export const handleValueCardVariables = (title, content, values, card) => {
```

Comment:
jsdoc here too - I know this seems redundant, but it's particularly important to highlight the nuance here between the logic for ValueCard variables vs ImageCard variables vs TableCard variables etc.